src/main/java/tech/gruppone/stalker/server/services/OrganizationService.java

```diff
@@ -1,72 +1,112 @@
+    final String organizationType = organization.getOrganizationType();
+    final Timestamp creationDateTime = Timestamp.valueOf(organization.getCreatedDate());
+    final Timestamp lastChangeDateTime = Timestamp.valueOf(organization.getLastModifiedDate());
+
+    final var organizationData =
+        OrganizationDataDto.builder()
+            .name(name)
+            .description(description)
+            .organizationType(organizationType)
+            .creationDateTime(creationDateTime)
+            .lastChangeDateTime(lastChangeDateTime)
+            .build();
+
+    return OrganizationDto.builder().id(id).data(organizationData).placeIds(placeIds).build();
+  }
 
   public Flux<OrganizationDto> findAll() {
-    // I have no idea why this works
-    return organizationRepository.findAll().map(OrganizationDao::getId).flatMap(this::findById);
+    final var organizationDaos = organizationRepository.findAll();
+    final var placeDaos = placeRepository.findAll();
+
+    return organizationDaos
+        .flatMap(
+            organizationDao -> {
+              final Flux<Long> placeIds =
+                  placeDaos
+                      .filter(
+                          placeDao -> placeDao.getOrganizationId().equals(organizationDao.getId()))
+                      .map(PlaceDao::getId);
+
+              return Mono.just(organizationDao).zipWith(placeIds.collectList());
+            })
+        .map(this::fromTuple);
   }
 
   public Mono<OrganizationDto> findById(final long id) {
-    final Flux<PlaceDto> places = placeService.findAllByOrganizationId(id);
     final Mono<OrganizationDao> organizationDao = organizationRepository.findById(id);
+    final Flux<Long> placeIdsFlux =
+        placeRepository.findAllByOrganizationId(id).map(PlaceDao::getId);
 
-    final OrganizationDataDtoBuilder builder = OrganizationDataDto.builder();
-    places.subscribe(builder::place);
-
-    return places
-        .then(organizationDao)
-        .map(
-            orgDao ->
-                builder
-                    .name(orgDao.getName())
-                    .description(orgDao.getDescription())
-                    .isPrivate(orgDao.getIsPrivate())
-                    .creationDateTime(Timestamp.valueOf(orgDao.getCreatedDate()))
-                    .lastChangeDateTime(Timestamp.valueOf(orgDao.getLastModifiedDate()))
-                    .build())
-        .map(data -> new OrganizationDto(id, data));
+    return organizationDao.zipWith(placeIdsFlux.collectList()).map(this::fromTuple);
   }
 
-  public Mono<Long> save(OrganizationDataDto organizationDataDto) {
+  private OrganizationDao fromDataDtoWithoutDates(final OrganizationDataDto organizationDataDto) {
+
+    final String name = organizationDataDto.getName();
+    final String description = organizationDataDto.getDescription();
+    final String organizationType = organizationDataDto.getOrganizationType();
+
+    return OrganizationDao.builder()
```

Comment:
Manca la possibilità di inserire i dati LDAP nell'organizzazione che stiamo andando a creare (ldapConfiguration)

Reply:
verrà gestito da un servizio a parte, visto che anche nel db è in una tabella a parte